pkg/controller/pcideviceclaim/pcideviceclaim_controller.go

```diff
@@ -46,67 +49,36 @@ func Register(
 		msg := fmt.Sprintf("cannot obtain KubeVirt client: %v", err)
 		return errors.New(msg)
 	}
+	nodename := os.Getenv("NODE_NAME")
 	handler := &Handler{
 		pdcClient:  pdcClient,
 		pdClient:   pdClient,
 		virtClient: virtClient,
+		nodeName:   nodename,
 	}
+
 	pdcClient.OnRemove(ctx, "PCIDeviceClaimOnRemove", handler.OnRemove)
-	pdcClient.OnChange(ctx, "PCIDeviceClaimOnChange", handler.OnChange)
-	nodename := os.Getenv("NODE_NAME")
-	handler.rebindAfterReboot(nodename)
-	// TODO fix stale PCIDevice CRs causing the Pod to crash
-	//err := handler.unbindOrphanedPCIDevices(nodename)
-	//if err != nil {
-	//	return err
-	//}
+	pdcClient.OnChange(ctx, "PCIDeviceClaimReconcile", handler.reconcilePCIDeviceClaims)
+	handler.rebindAfterReboot()
+	err = handler.unbindOrphanedPCIDevices()
+	if err != nil {
+		return err
+	}
 	// Load VFIO drivers when controller starts instead of repeatedly in the reconcile loop
 	loadVfioDrivers()
-
-	// start goroutine to regularly reconcile the PCI Device Claims' status with their spec
-	go func() {
-		ticker := time.NewTicker(reconcilePeriod)
-		for range ticker.C {
-			logrus.Info("Reconciling PCI Device Claims list")
-			if err := handler.reconcilePCIDeviceClaims(nodename); err != nil {
-				logrus.Errorf("PCI Device Claim reconciliation error: %v", err)
-			}
-		}
-	}()
 	return nil
 }
 
-func (h Handler) OnChange(name string, pdc *v1beta1.PCIDeviceClaim) (*v1beta1.PCIDeviceClaim, error) {
-	if pdc != nil && pdc.Spec.NodeName != os.Getenv("NODE_NAME") {
-		return pdc, nil
-	}
-	_, err := h.pdcClient.Get(name, v1.GetOptions{})
-	if err != nil {
-		return pdc, err
-	}
-	// Equivalent to OnCreate, since the PCIDeviceClaim doesn't exist yet
-	err = h.attemptToEnablePassthrough(pdc)
-	if err != nil {
-		return pdc, err
-	}
-
-	return pdc, nil
-}
-
 // When a PCIDeviceClaim is removed, we need to unbind the device from the vfio-pci driver
 func (h Handler) OnRemove(name string, pdc *v1beta1.PCIDeviceClaim) (*v1beta1.PCIDeviceClaim, error) {
-	if pdc.Spec.NodeName != os.Getenv("NODE_NAME") {
+	if pdc == nil || pdc.DeletionTimestamp == nil || pdc.Spec.NodeName != h.nodeName {
 		return pdc, nil
 	}
+	logrus.Infof("OnRemove(%s, %s)", name, pdc.Name)
```

Comment:
please put the `logrus.Info` behind the check of `pdc==nil`, as `pdc.Name` is used in logrus.